src/qt/listbox.cpp

```diff
@@ -21,29 +21,49 @@ class wxQtListWidget : public wxQtEventSignalHandler< QListWidget, wxListBox >
 private:
     void clicked( const QModelIndex &index );
     void doubleClicked( const QModelIndex &index );
+    void itemChanged(QListWidgetItem *item);
 };
 
 wxQtListWidget::wxQtListWidget( wxWindow *parent, wxListBox *handler )
     : wxQtEventSignalHandler< QListWidget, wxListBox >( parent, handler )
 {
     connect(this, &QListWidget::clicked, this, &wxQtListWidget::clicked);
     connect(this, &QListWidget::doubleClicked, this, &wxQtListWidget::doubleClicked);
+    connect(this, &QListWidget::itemChanged, this, &wxQtListWidget::itemChanged);
 }
 
 void wxQtListWidget::clicked(const QModelIndex &index )
 {
     wxListBox *handler = GetHandler();
     if ( handler )
-        handler->QtSendEvent(wxEVT_LISTBOX, index, true);
+        handler->QtSendEvent(wxEVT_LISTBOX, index.row(), true);
 }
 
 void wxQtListWidget::doubleClicked( const QModelIndex &index )
 {
     wxListBox *handler = GetHandler();
     if ( handler )
-        handler->QtSendEvent(wxEVT_LISTBOX_DCLICK, index, true);
+        handler->QtSendEvent(wxEVT_LISTBOX_DCLICK, index.row(), true);
 }
 
+void wxQtListWidget::itemChanged(QListWidgetItem *item)
+{
+    if (item->flags() & Qt::ItemIsUserCheckable)
+    {
+        wxListBox *handler = GetHandler();
+        if (handler)
+        {
+            QWidget *widgetHandle = handler->GetHandle();
+            QListWidget *listWidget = dynamic_cast<QListWidget*>(widgetHandle);
```

Comment:
Is there any other QWidget that can possibly be wrapped by `wxListBox`?

Reply:
There's `QListView `that could be used, although it's almost identical except in how it binds to data

Reply:
Hold on, I'm getting really confused here with all these `handle`s and `handler`s.
We're in a `wxQtListWidget` here, which derives from `QListWidget`, which pretty much makes `listWidget` == `this`, doesn't it? So are all the `GetHandler()`, `GetHandle()`, `dynamic_cast()` and `if()`s really needed?

Reply:
True, the `getHandle` and `dynamic_cast` could be removed and just call `this->row(item)` instead. I think the `GetHandler` is still needed in order to call `QtSendEvent`